### ISSUE TITLE
Question about Adjacencies & Inversion

### ISSUE THREAD

**Issue by greg-sharpe** [ga4gh/vrs#469]
I have a question about how to specify AdjoiningSequences in an Adjacency in the case of an inversion.

![image](https://github.com/ga4gh/vrs/assets/109313554/f8f287d8-5c3e-4be9-b143-b2859532bdf4)

Is it possible to define AdjoiningSequence 2 the same way in each Adjacency, so that AdjoiningSequence 2 could be a single SequenceLocation object referenced by both Adjacencies?

The problem I'm seeing is that the definition of SequenceLocation requires End greater than or equal to Start, meaning that AdjoiningSequence 2 needs to have {Start: C, End: D}. Then it would be impossible to look at either Adjacency and determine that it's describing an inversion, as opposed to describing an adjacency that's present in the reference sequence

(Edited per Larry's correction that End >= Start)

**Comment by larrybabb** [comment]
I'll defer to @ahwagner and @d-cameron but I believe we intentionally modified the definition of SequenceLocation so that "start can now be greater than end" (I believe you phrased it backwards in your post above). We did that specifically to support the ability to address inversions and other reverse complemented sequences based on a reference.  Our documentation for SequenceLocation has not been updated to reflect that yet.

@ahwagner can you verify this?

**Comment by larrybabb** [comment]
In your example above I would refer to two examples that we have images for in the repo currently [Reverse complement adjacency example](https://github.com/ga4gh/vrs/blob/2.x/docs/source/images/ex_revcomp_breakpoint.png) and [SV Haplotype example](https://github.com/ga4gh/vrs/blob/2.x/docs/source/images/ex_sv_haplotype.png).

in the reverse complement adjacency it illustrates that fact that you only provide the `start` or the `end` when specifying either of the SequenceLocations in the `adjoinedSequences` (aka breakpoint).  When you provide a `start` you are saying that that side of the sequence decreases as it moves away from the adjacency. And likewise when you provide an `end` you are saying that that side of the adjoined sequence increases as it moves away from the adjacency. 

In the Reverse complement adjacency example the 2nd sequenceLocation specifies the `end` which readily identifies that the second sequence is reverse complemented.   

You could apply this to the SV Haplotype example by simply changing the `start` and `end` attributes from the 2nd part of the first adjacency with the 1st part of the second adjacency.

@ahwagner please verify my response...

**Comment by larrybabb** [comment]
I don't think we have "yet" changed our requirement that End >= Start when fully specifying a SequenceLocation. If this is true (and I'll wait on @ahwagner 's validation). That means any inversion would require the use of adjacencies to be defined.

**Comment by ahwagner** [comment]
In the current model, the `Adjacency` only describes the sequence junction, nothing beyond. So the use of `A` and `F` are not allowed as described in the above model; instead you would have 4 Adjacency objects, ordered in a containing structure (the current proposal is that structure is a revised `Haplotype`, but there is an open discussion about that at #461). The information for the four adjacency objects, per the current [schema](https://github.com/ga4gh/vrs/blob/62271fe053be437d6238b42bd9d40f1a576fa5cf/schema/vrs.yaml#L565-L630) and [examples](https://github.com/ga4gh/vrs/blob/62271fe053be437d6238b42bd9d40f1a576fa5cf/tests/test_definitions.yaml#L2-L35) would look like:

Adjacency 1: [{start: A}] (if terminal)
Adjacency 2: [{end: B}, {end: D}]
Adjacency 3: [{start: C}, {start: E}]
Adjacency 4: [{end: F}] (if terminal)

Hope this helps.

**Comment by greg-sharpe** [comment]
Thanks for the clarification!

**Comment by Mrinal-Thomas-Epic** [comment]
@ahwagner @larrybabb Is the constraint (that only one of `start` or `stop` can be specified on a `SequenceLocation` within an `Adjacency` specified explicitly anywhere in the schema)?

**Comment by ahwagner** [comment]
@Mrinal-Thomas-Epic no, there is no constraint specified at the schema level; just the bare-bones work at the moment. I can't get to creating this constraint in the near term, but I think it would be useful to have it. If you agree, please make a separate issue requesting this feature.
